packages/itwin/grouping-mapping-widget/src/components/Mappings/hooks/useMappingsOperations.ts

```diff
@@ -33,29 +36,25 @@ const fetchExtractionStatus = async (
   iModelId: string,
   getAccessToken: GetAccessTokenFn,
   extractionClient: IExtractionClient,
-  setExtractionIconData: React.Dispatch<React.SetStateAction<ExtractionIconData>>,
-  setExtractionMessageData: React.Dispatch<React.SetStateAction<ExtractionMessageData[]>>
+  setExtractionStatusIcon: (extractionStatusIcon: ExtractionStatusData | ((extractionStatusIcon: ExtractionStatusData) => ExtractionStatusData)) => void,
+  setExtractionMessageData: (extractionMessageData: ExtractionMessageData[] | ((extractionMessageData: ExtractionMessageData[]) => ExtractionMessageData[])) => void
 ) => {
-  setExtractionIconData({
-    iconStatus: "warning",
-    iconMessage: "Extraction status pending.",
-  });
   try {
+    setExtractionStatusIcon({
+      iconStatus: undefined,
+      iconMessage: "Loading...",
+    });
     const accessToken = await getAccessToken();
-    const extractions = await extractionClient.getExtractionHistory(accessToken, iModelId, 1);
-    if (extractions.length === 0) {
-      setExtractionIconData({
+    const extraction = await extractionClient.getExtractionHistory(accessToken, iModelId, 1);
```

Comment:
Use `getExtractionHistoryIterator` here with `top = 1` and take only the first history entry. Using `getExtractionHistory` causes the whole history to be loaded in pages of size 1 (top = 1).

Using `getExtractionHistoryIterator` should produce better results, since it loads pages as you iterate, so if you take only 1 element, it will query only once.

Reply:
I have changed it up to `getExtractionHistoryIterator`.